pkg/git/github/download.go

```diff
@@ -81,6 +84,40 @@ func (f *FileService) DownloadFile(org, repo, path, branch string) (string, erro
 	return contents, nil
 }
 
+func (f *FileService) DownloadContents(org, repo, path, branch string) (string, error) {
+	client := http.DefaultClient
+	req, err := http.NewRequest("GET", f.EncodeURL(org, repo, path, branch), nil)
+	if err != nil {
+		return "", err
+	}
+	if token := f.GitHub.GetToken(); token != "" {
+		req.Header.Add("Authorization", fmt.Sprintf("token %s", f.GitHub.GetToken()))
+	}
+	req.Header.Add("Accept", "application/vnd.github.v3.raw")
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+
+	//check GitHub response
+	if err = github.CheckResponse(resp); err != nil {
```

Comment:
What do the errors look like here? Wondering if we could provide some more context for end users if we fail to download from GitHub.

Reply:
Yes, indeed this is the content of `CheckResponse` from the Github library [google/go-github/v33/github/github.go](https://github.com/armory/dinghy/blob/master/vendor/github.com/google/go-github/v33/github/github.go#L781-L823)

This is abstracting the error for us, however, I added a struct definition that we had in the previous DownloadContent method.